src/Entities/Boat.h

```diff
@@ -31,8 +31,27 @@ class cBoat :
 	virtual void HandleSpeedFromAttachee(float a_Forward, float a_Sideways) override;
 
 	cBoat(double a_X, double a_Y, double a_Z);
-} ;
 
+	int GetTimeSinceLastHit(void) const { return m_TimeSinceLastHit; }
```

Comment:
I've changed time since last hit to follow the model of minecarts which used the name "LastDamage"

I've also set LastDamage to 10 upon taking damage and decrement it by 1 each tick, per Pokechu22's recommendation.